PP-1089-1090: Skipping tests instead of failing them when no.of MoMs …

prakashcv13 opened the pull request:
…hosts provided is not as per the expectations of the test.

<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Issue-ID
<!--- Replace XXXX below (2 places) with the actual JIRA id -->
* **[PP-1089](https://pbspro.atlassian.net/browse/PP-1089)**
* **[PP-1090](https://pbspro.atlassian.net/browse/PP-1090)**

#### Problem description
* TestJobRequeueTimeoutErrorMsg and TestOnlySmallFilesOverTPP were asserting if the no.of MoM hosts provided was not as expected by the test instead of skipping.

#### Cause / Analysis
* The two test suites need to be skipped if the no.of MoM hosts is not as per the expectations.

#### Solution description
* Changed the code to skip instead of assert.

#### Checklist:
<!--- Use the preview button to see the checkboxes/links properly. -->
<!--- Go over the following points, and put an `x` (without spaces around it) in the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.


__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

anamikau commented on test/tests/functional/pbs_job_requeue_timeout_error.py:
```diff
@@ -50,9 +50,8 @@ def setUp(self):
-                              '  use -p moms=<mom1>:<mom2>')
-            self.assertEqual(len(self.moms), 2)
+            self.skip_test(reason="need 2 mom hosts: -p moms=<m1>:<m2>")
+            return
```
you do not need a return. skip_test will do the needful

anamikau commented on test/tests/functional/pbs_only_small_files_over_tpp.py:
```diff
@@ -50,9 +50,8 @@ def setUp(self):
-                              '  use -p moms=<mom1>:<mom2>')
-            self.assertEqual(len(self.moms), 2)
+            self.skip_test(reason="need 2 mom hosts: -p moms=<m1>:<m2>")
+            return
```
you do not need a return. skip_test will do the needful

prakashcv13 commented:
@anamikau, I have removed the redundant return calls. However, there are other files that have them. pbspro/test/tests/functional$ grep -A 1 -n skip_test * | grep return
grep: cgroups: Is a directory
pbs_hook_execjob_prologue.py-52-            return
pbs_node_rampdown.py-274-            return
pbs_power_provisioning_sgi.py-219-            return

anamikau commented:
@prakashcv13 ,thanks for pointing this out. Looks like they were missed during reviews.

prakashcv13 commented:
@anamikau, I am ok with removing the redundant return calls from the 3 files I found, so I went ahead and removed them. You might have to re-approve it.

prakashcv13 commented:
@anamikau, I tried executing the test suite from pbs_power_provisioning_sgi.py, the test errors out instead of skip. So, I will need to spend some time to see if there is an error in the script or we need a specific configuration needed so that the test gets skipped. That will be an overkill for this change as per my understanding. So, we have two options - 
1) We go ahead and merge these changes.
2) I would back out the changes from pbs_power_provisioning_sgi.py and then we merge the remaining changes.

I leave the choice to you.

anamikau commented:
@prakashcv13 since the changes has been approved and code fix is not trivial we can go with the fix. But please file a bug for pbs_power_provisioning_sgi.py and mention the issue ID in your JIRA ticket as why there are no results for power provisioning tests.

prakashcv13 commented:
@anamikau, I feel we are unnecessarily spending more time than we should for this trivial change. I have already mentioned spending more time to get the logs for OR providing a reason to not work on it for this change seems like an overkill to me. I have removed the changes I did to pbs_power_provisioning_sgi.py and squashed all my commits. 

I will rebase once a maintainer is ready for merge.
Thank you!

bhroam commented:
@prakashcv13 Go ahead and rebase and I'll merge your PR.

prakashcv13 commented:
@bhroam, Thank you, I just rebased my branch.

bhroam commented:
I merged in the PR.

prakashcv13 commented:
thank you @bhroam.
